src/gui/mzroll/peakdetectiondialog.cpp

```diff
@@ -417,6 +506,62 @@ void PeakDetectionDialog::inputInitialValuesPeakDetectionDialog() {
     }
 }
 
+void PeakDetectionDialog::getModelsList()
+{
+    QStringList args;
+    _modelsList.clear();
+    auto _dlManager = new DownloadManager();
+    auto _pollyIntegration = new PollyIntegration(_dlManager);
+
+    auto cookieFile = QStandardPaths::writableLocation(
+                    QStandardPaths::GenericConfigLocation)
+                    + QDir::separator() 
+                    + "El-MAVEN_cookie.json" ;
+    
+    ifstream readCookie(cookieFile.toStdString());
+    if(!readCookie.is_open())
+        return;
+
+    json cookieInput = json::parse(readCookie);
+    string refreshToken = cookieInput["refreshToken"];
+    string idToken = cookieInput["idToken"];
+
+    QString cookies;
+    cookies += "refreshToken=";
+    cookies += QString::fromStdString(refreshToken);
+    cookies += ";";
+    cookies += "idToken=";
+    cookies += QString::fromStdString(idToken);
+
+    args << cookies;
+
+    auto res = _pollyIntegration->runQtProcess("listBucketObjects", args);
+    QString str(res[0].constData());
+    if (str.isEmpty())
+        return;
+    auto splitString = mzUtils::split(str.toStdString(), "\n");
+    auto data = splitString[splitString.size() - 2];
+    auto dataQstring = QString::fromStdString(data);
+    QJsonObject dataObj;
+
+    QJsonDocument doc = QJsonDocument::fromJson(dataQstring.toUtf8());
+    dataObj = doc.object();  
```

Comment:
I remember seeing some very similar logic in `BackgroundOpsThread` class. Is it possible to share it? Anyhow, you should consider moving this whole work to the background thread. Reading and parsing data is a potentially blocking operation and should not be part of the main thread. Instead you could show a status string here (which I think happens post-detection when peaks are about to be classified). Or you could totally ignore all of this if you are sure that the files will never be large enough to cause an issue.

Reply:
The reason to use the same logic here is that we need to change the content of the dropdown (select models) on the basis of the Polly credentials. This is done so that the user can choose one model out of many to run PeakML.

Reply:
Hm, yes and the main "fetch model list" logic can still be moved to `BackgroundOpsThread` to keep all such methods in one place. But it's up to you. I do not mind repetitions in two places (I would if it was repeated thrice).